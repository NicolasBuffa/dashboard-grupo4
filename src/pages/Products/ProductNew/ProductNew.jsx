import "../ProductView/ProductView.css";
import { useEffect } from "react";
import { useForm } from "../../../hooks/useForm";
import { useGetOneProduct } from "../../../hooks/useGetOneProduct";

import { useParams } from "react-router-dom";

import FormNewProduct from "../../../components/Form/FormNewProduct";

import HeaderContent from "../../../components/Header/HeaderContent/HeaderContent";
import Header from "../../../components/Header/Header";

let initialForm = {
  title: "",
  price: 0,
  stock: 0,
  description: "",
  category: "",
  images: [],
  rating: {
    rate: 0,
    count: 0,
  },
};

function ProductNew() {
  const { product } = useGetOneProduct(initialForm);

  const params = useParams();

  const {
    onInputChange,
    handdleAddImage,
    handleSubmit,
    handleSubmitNewProduct,
    handdleDelete,
    handleDecrement,
    handleIncrement,
    onResetForm,
    // Valores del formState
    formState,
    price,
    description,
    title,
    stock,
    images,
  } = useForm(initialForm);
  useEffect(() => {
    console.log(formState);
  }, [formState]);

  return (
    <div className="productView-container">
      <Header>
        <HeaderContent>
          <h1>Productos &#62; Nuevo Producto</h1>
        </HeaderContent>
        <HeaderContent></HeaderContent>
      </Header>

      {/* Esto es un componente con la info del producto */}
      <div className="productView-container_detail">
        <div className="productView-container_detailContainerMain"></div>
        {/* esto es un componente con un use form */}
        <div className="productView-container_formContainer">
          <h3>Informacion</h3>
          <FormNewProduct product={product} />
        </div>
      </div>
    </div>
  );
}

export default ProductNew;
