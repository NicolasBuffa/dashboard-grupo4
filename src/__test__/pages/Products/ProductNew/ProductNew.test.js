import ProductNew from "../../../../pages/Products/ProductNew/ProductNew";
import { render, screen } from "@testing-library/react";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => jest.fn(),
}));

describe("TEST productNre", () => {
  test("debe hacer match con el snapShot", () => {
    const { container } = render(<ProductNew />);
    expect(container).toMatchSnapshot();
  });
});
