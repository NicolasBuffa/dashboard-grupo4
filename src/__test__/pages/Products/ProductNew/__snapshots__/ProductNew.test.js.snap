// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST productNre debe hacer match con el snapShot 1`] = `
<div>
  <div
    class="productView-container"
  >
    <header
      class="Header"
    >
      <div
        class="HeaderContent"
      >
        <h2>
          <a
            href="/products"
          >
            Productos 
          </a>
          &gt; Nuevo Producto
        </h2>
         
      </div>
      <div
        class="HeaderContent"
      />
    </header>
    <div
      class="productView-container_detail"
    >
      <div
        class="productView-container_detailContainerMain"
      />
      <div
        class="productView-container_formContainer"
      >
        <h3>
          Informacion
        </h3>
        <form
          action="submit"
          role="form"
        >
          <label>
            Nombre
          </label>
          <input
            class="inputForm"
            name="title"
            placeholder="inputValue"
            role="title"
            type="text"
            value=""
          />
          <label>
            Valor
          </label>
          <input
            class="inputForm"
            name="price"
            placeholder="inputValue"
            role="price"
            type="number"
            value="0"
          />
          <label>
            Stock
          </label>
          <div>
            <button
              class="buttonStock"
              type="button"
            >
              -
            </button>
            <input
              class="inputForm"
              name="stock"
              placeholder="inputValue"
              role="stock"
              type="number"
              value="0"
            />
            <button
              class="buttonStock"
              type="button"
            >
              +
            </button>
          </div>
          <label>
            Description
          </label>
          <textarea
            class="inputForm"
            name="description"
            placeholder="inputValue"
            rows="12"
            type="text"
          />
          <label>
            Tienda
          </label>
          <select
            class="inputForm"
            id=""
            name=""
          >
            <option
              value="tienda1"
            >
              Olivia Store
            </option>
            <option
              selected=""
              value="store2"
            >
              Store 2
            </option>
            <option
              value="store3"
            >
              Store 3
            </option>
            <option
              value="store4"
            >
              Store 4
            </option>
          </select>
          <h3>
            Galeria de Imagenes
          </h3>
          <label>
            Nueva Imagen
          </label>
          <div
            class="containerUploadImages"
          >
            <input
              class="inputForm "
              id="inputImg"
              name="images"
              type="text"
            />
            <button
              class="subirImage"
              type="button"
            >
              SUBIR IMAGEN
            </button>
          </div>
          <div
            class="productView-container_actualImg"
          >
            <h5>
              Imagenes Actuales
            </h5>
          </div>
          <div
            class="containerSubmitButton"
          >
            <button
              class="submitButton"
              type="button"
            >
              Cancelar
            </button>
            <button
              class="submitButton"
              role="submmitButton"
            >
              Guardar
               
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
